packages/bridge/src/vl/utils/conversion/category/index.ts

```diff
@@ -10,11 +10,16 @@ import { findColorForCategory } from '../histogram';
  * @returns {object[]}
  */
 export function vlToCategory(histogram: VLCategoricalHistogram, legendData: any = []): object[] {
-  return histogram.value.map((d) => ({
-    color: findColorForCategory(d.x, legendData),
-    name: d.x,
-    value: d.y
-  }));
+  const fullCategories = (histogram as any)._categories;
+  const values = histogram.value;
+  return fullCategories.map((d) => {
+    const value = values.find((elem) => elem.x === d.name);
```

Comment:
Just a performance quirk, it'd be better to set an object with category name as a key and the value as the actual object value. That way we will just iterate once over all object.

Reply:
Totally, my fault!